### PR TITLE
DateTimeFormatterPreferences

### PR DESCRIPTION
**Pull request by sffc**
#1317

### CODE REVIEW

File: components/calendar/src/any_calendar.rs [hunk 1]
```diff
@@ -748,13 +762,13 @@ impl AnyCalendar {
     ///
     /// [📚 Help choosing a constructor](icu_provider::constructors)
     #[cfg(feature = "compiled_data")]
-    pub fn new_for_locale(locale: &DataLocale) -> Self {
-        let kind = AnyCalendarKind::from_data_locale_with_fallback(locale);
+    pub fn new_for_locale(prefs: AnyCalendarPreferences) -> Self {
```
Comment (sffc) [inline components/calendar/src/any_calendar.rs:765]: Should I rename this function?

Comment (zbraniecki) [reply components/calendar/src/any_calendar.rs:765]: I think so - just `new`?

File: provider/core/src/marker.rs [hunk 2]
```diff
@@ -603,6 +603,14 @@ impl DataMarkerInfo {
             Err(DataErrorKind::MarkerNotFound.with_marker(marker))
         }
     }
+
+    /// Constructs a [`DataLocale`] for this [`DataMarkerInfo`].
+    pub fn make_locale(
```
Comment (sffc) [inline provider/core/src/marker.rs:608]: Notice the new function in `icu_provider`. OK?

**Review by zbraniecki — APPROVED** [review]
this feels so easy to read. Looks great!

**Review by Manishearth — DISMISSED** [review]

**Review by zbraniecki — DISMISSED** [review]

File: components/calendar/src/any_calendar.rs (hunk 1 above)
Comment (sffc) [reply components/calendar/src/any_calendar.rs:765]: @Manishearth What do you think about the name of this function? (a constructor for AnyCalendar)

File: components/calendar/src/any_calendar.rs [hunk 3]
```diff
@@ -21,12 +21,26 @@ use crate::roc::Roc;
...
 
 use core::fmt;
 
+define_preferences!(
+    /// The prefs for datetime formatting.
+    [Copy]
+    AnyCalendarPreferences,
+    {
+        /// The user's preferred calendar system.
+        calendar: CalendarAlgorithm
```
[the remark this answers is not in the document]
Comment (sffc) [reply components/calendar/src/any_calendar.rs:40]: I guess I'll stick with the UTS 35 language: "calendar algorithm"

https://unicode.org/reports/tr35/tr35.html#Key_And_Type_Definitions_

File: components/calendar/src/any_calendar.rs (hunk 1 above)
Comment (sffc) [reply components/calendar/src/any_calendar.rs:765]: We already have

```rust
    #[cfg(feature = "compiled_data")]
    pub const fn new(kind: AnyCalendarKind) -> Self
```

File: components/calendar/src/any_calendar.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Manishearth) [reply components/calendar/src/any_calendar.rs:40]: spec name works for me.

File: components/calendar/src/any_calendar.rs (hunk 1 above)
Comment (Manishearth) [reply components/calendar/src/any_calendar.rs:765]: I'm fine with that being renamed if we need. `new_for_kind(kind)` vs `new(prefs)` or `new(kind)` vs `new_for_prefs(prefs)`/`new_for_locale(prefs)`

Comment (sffc) [reply components/calendar/src/any_calendar.rs:765]: I made https://github.com/unicode-org/icu4x/issues/5843 to discuss this before 2.0 (but after 2.0 Beta) and unblock this PR

**Review by Manishearth — APPROVED** [review]

**Pull request merged**